created Drivetrain

TurtleMinecraft opened the pull request.

Commit:
created Drivetrain

OfriRosenbaum commented on src/main/java/frc/robot/OI.java:
```diff
@@ -1,8 +1,19 @@
+        return -right.getY();
+    }
+
+    public double getLeftX() {
```
left before right

OfriRosenbaum commented on src/main/java/frc/robot/OI.java:
```diff
@@ -1,8 +1,19 @@
+    private final Joystick right = new Joystick(1);
+    private final Joystick left = new Joystick(0);
 
+    public double getRightY() {
```
right after left

OfriRosenbaum commented on src/main/java/frc/robot/OI.java:
```diff
@@ -1,8 +1,19 @@
 package frc.robot;
 
-public class OI /*GEVALD*/{
+import com.spikes2212.util.XboxControllerWrapper;
```
remove redundant import

OfriRosenbaum commented on src/main/java/frc/robot/OI.java:
```diff
@@ -1,8 +1,19 @@
 
-    public OI() {
+public class OI /*GEVALD*/ {
+    private final Joystick right = new Joystick(1);
```
blank line at the start of the class

OfriRosenbaum replied:
right after left

OfriRosenbaum commented on src/main/java/frc/robot/OI.java:
```diff
@@ -1,8 +1,19 @@
-    public OI() {
+public class OI /*GEVALD*/ {
+    private final Joystick right = new Joystick(1);
+    private final Joystick left = new Joystick(0);
```
left before right

OfriRosenbaum commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -15,77 +18,102 @@
-  /** This function is called periodically whilst in simulation. */
-  @Override
-  public void simulationPeriodic() {}
+    Drivetrain drivetrain;
```
private??

OfriRosenbaum commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -15,77 +18,102 @@
-  @Override
-  public void simulationPeriodic() {}
+    Drivetrain drivetrain;
+    OI oi;
```
ditto

OfriRosenbaum commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -4,8 +4,11 @@
 
 package frc.robot;
 
+import com.spikes2212.command.drivetrains.TankDrivetrain;
```
remove redundant import

OfriRosenbaum commented on src/main/java/frc/robot/RobotMap.java:
```diff
@@ -4,6 +4,11 @@ public class RobotMap {
 
     public interface CAN {
 
+        int DRIVETRAIN_LEFT_NEO_1 = 1;
```
we name the ports after the controller name, not the motor (don't write `NEO`, write `SPARK_MAX`)

OfriRosenbaum replied:
initial ports should be -1, unless known the exact correct ports

OfriRosenbaum commented on src/main/java/frc/robot/RobotMap.java:
```diff
@@ -4,6 +4,11 @@ public class RobotMap {
     public interface CAN {
 
+        int DRIVETRAIN_LEFT_NEO_1 = 1;
+        int DRIVETRAIN_LEFT_NEO_2 = 2;
```
ditto

OfriRosenbaum commented on src/main/java/frc/robot/RobotMap.java:
```diff
@@ -4,6 +4,11 @@ public class RobotMap {
 
+        int DRIVETRAIN_LEFT_NEO_1 = 1;
+        int DRIVETRAIN_LEFT_NEO_2 = 2;
+        int DRIVETRAIN_RIGHT_NEO_1 = 3;
```
ditto

OfriRosenbaum commented on src/main/java/frc/robot/RobotMap.java:
```diff
@@ -4,6 +4,11 @@ public class RobotMap {
+        int DRIVETRAIN_LEFT_NEO_1 = 1;
+        int DRIVETRAIN_LEFT_NEO_2 = 2;
+        int DRIVETRAIN_RIGHT_NEO_1 = 3;
+        int DRIVETRAIN_RIGHT_NEO_2 = 4;
```
![image](https://user-images.githubusercontent.com/71877371/189051088-30e5fbed-8062-457f-ac11-c0a5db60db2f.png)

OfriRosenbaum commented on src/main/java/frc/robot/subsystems/Drivetrain.java:
```diff
@@ -0,0 +1,28 @@
+import edu.wpi.first.wpilibj.ADXRS450_Gyro;
+import edu.wpi.first.wpilibj.motorcontrol.MotorController;
+import edu.wpi.first.wpilibj.motorcontrol.MotorControllerGroup;
+import frc.robot.Robot;
```
remove all redundant imports (`CTRL + ALT + O`)

OfriRosenbaum commented on src/main/java/frc/robot/subsystems/Drivetrain.java:
```diff
@@ -0,0 +1,28 @@
+
+public class Drivetrain extends TankDrivetrain {
+
+    ADXRS450_Gyro gyro;
```
private?

Commit:
made Drivetrain singleton

OfriRosenbaum reviewed (changes requested):
CTRL + ALT + L

Commit:
fixed OI class ordering

Commit:
fixed OI class ordering
fixed Robot class ordering

Commit:
fixed OI class ordering
fixed Robot class ordering
fixed RobotMap class ordering
fixed Drivetrain class ordering

OfriRosenbaum commented on src/main/java/frc/robot/RobotMap.java:
```diff
@@ -4,6 +4,11 @@ public class RobotMap {
+        int DRIVETRAIN_LEFT_SPARKMAX_2 = -1;
+        int DRIVETRAIN_RIGHT_SPARKMAX_1 = -1;
+        int DRIVETRAIN_RIGHT_SPARKMAX_2 = -1;
+
```
redundant blank line

OfriRosenbaum reviewed (changes requested):
`CTRL + ALT + L` in every file you have created

OfriRosenbaum commented on src/main/java/frc/robot/subsystems/Drivetrain.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    private Drivetrain() {
+        super(new MotorControllerGroup(new CANSparkMax(RobotMap.CAN.DRIVETRAIN_LEFT_NEO_1, CANSparkMaxLowLevel.MotorType.kBrushless),
```
you changed the name of the ports in `RobotMap` but not here, change `NEO` to `SPARKMAX`

OfriRosenbaum commented on src/main/java/frc/robot/subsystems/Drivetrain.java:
```diff
@@ -0,0 +1,35 @@
+
+    private Drivetrain() {
+        super(new MotorControllerGroup(new CANSparkMax(RobotMap.CAN.DRIVETRAIN_LEFT_NEO_1, CANSparkMaxLowLevel.MotorType.kBrushless),
+                new CANSparkMax(RobotMap.CAN.DRIVETRAIN_LEFT_NEO_2, CANSparkMaxLowLevel.MotorType.kBrushless)),
```
ditto

OfriRosenbaum commented on src/main/java/frc/robot/subsystems/Drivetrain.java:
```diff
@@ -0,0 +1,35 @@
+    private Drivetrain() {
+        super(new MotorControllerGroup(new CANSparkMax(RobotMap.CAN.DRIVETRAIN_LEFT_NEO_1, CANSparkMaxLowLevel.MotorType.kBrushless),
+                new CANSparkMax(RobotMap.CAN.DRIVETRAIN_LEFT_NEO_2, CANSparkMaxLowLevel.MotorType.kBrushless)),
+                new MotorControllerGroup(new CANSparkMax(RobotMap.CAN.DRIVETRAIN_RIGHT_NEO_1, CANSparkMaxLowLevel.MotorType.kBrushless),
```
ditto

OfriRosenbaum commented on src/main/java/frc/robot/subsystems/Drivetrain.java:
```diff
@@ -0,0 +1,35 @@
+        super(new MotorControllerGroup(new CANSparkMax(RobotMap.CAN.DRIVETRAIN_LEFT_NEO_1, CANSparkMaxLowLevel.MotorType.kBrushless),
+                new CANSparkMax(RobotMap.CAN.DRIVETRAIN_LEFT_NEO_2, CANSparkMaxLowLevel.MotorType.kBrushless)),
+                new MotorControllerGroup(new CANSparkMax(RobotMap.CAN.DRIVETRAIN_RIGHT_NEO_1, CANSparkMaxLowLevel.MotorType.kBrushless),
+                new CANSparkMax(RobotMap.CAN.DRIVETRAIN_RIGHT_NEO_2, CANSparkMaxLowLevel.MotorType.kBrushless)));
```
![image](https://user-images.githubusercontent.com/71877371/189514180-6e9f7d88-c6d8-4697-9637-76add4397484.png)

OfriRosenbaum commented on src/main/java/frc/robot/subsystems/Drivetrain.java:
```diff
@@ -0,0 +1,35 @@
+
+    private ADXRS450_Gyro gyro;
+
+    private static Drivetrain instance;
```
static comes before non-static

Commit:
fixed RobotMap class ordering
fixed Drivetrain class ordering
